src/main/java/com/microsoft/aad/msal4j/SilentParameters.java

```diff
@@ -88,8 +88,21 @@ public static SilentParametersBuilder builder(Set<String> scopes, IAccount accou
      */
     @Deprecated
     public static SilentParametersBuilder builder(Set<String> scopes) {
-        validateNotEmpty("scopes", scopes);
+        validateNotNull("scopes", scopes);
 
-        return builder().scopes(scopes);
+        return builder().scopes(removeEmptyScope(scopes));
+    }
+
+    private static Set<String> removeEmptyScope(Set<String> scopes){
+        // empty string is not a valid scope, but we currently accept it and can't remove support
+        // for it yet as its a breaking change. This will be removed eventually (throwing
+        // exception if empty scope is passed in).
+        Set<String> updatedScopes = new HashSet<>();
+        for(String scope: scopes){
+            if(!scope.equalsIgnoreCase(StringHelper.EMPTY_STRING)){
```

Comment:
trim might be also used to remove trailing and leading whitespaces

Reply:
Good point, I will add trim here.